### PR TITLE
Feature/int pow fast

### PR DESCRIPTION
**Pull request by serge-sans-paille**

### CODE REVIEW

**Comment by wolfv** [comment]
the build failures seem legit. We've made overloads in xtensor:

https://github.com/QuantStack/xtensor/blob/617f270ab5071e5a27be86b6cab3249e772233dd/include/xtensor/xmath.hpp#L330-L334

File: include/xsimd/math/xsimd_power.hpp [hunk 1]
```diff
@@ -85,6 +85,25 @@ namespace xsimd
         return detail::pow_kernel<batch<T, N>>::compute(x, y);
     }
 
+    template <class T0, class T1>
+    inline typename std::enable_if<std::is_integral<T1>::value, T0>::type
+    pow(T0 const &t0, T1 const &t1)
+    {
+      T0 a = t0;
```
Comment (wolfv) [inline include/xsimd/math/xsimd_power.hpp]: Just nitpicks:

We usually use 4 space indentation

also `const T0& t0`

File: include/xsimd/math/xsimd_power.hpp [hunk 2]
```diff
@@ -85,6 +85,25 @@ namespace xsimd
...
+      T1 b = t1;
+      bool const recip = b < 0;
+      T0 r{static_cast<T0>(1)};
+      while (1) {
```
Comment (wolfv) [inline include/xsimd/math/xsimd_power.hpp]: always brackets, and brackets on newline, e.g.

```
while(1)
{
    if(a & 1)
    {
          a++;
     }
}

```

File: test/xsimd_test_utils.hpp [hunk 3]
```diff
@@ -24,6 +24,50 @@
 #include "xtl/xcomplex.hpp"
 #endif
 
+// define some overloads here as complex version does not exist in std::
+// and integer version does not exist for msvc
+
+namespace utils {
+template <class T>
```
Comment (wolfv) [inline test/xsimd_test_utils.hpp]: namespace utils followed by newline + bracket. 

indentation inside namespace

File: test/xsimd_test_utils.hpp [hunk 4]
```diff
@@ -87,6 +131,13 @@ namespace xsimd
                 return lhs == rhs;
             }
 
+            if (utils::isnan(lhs)) {
+              return utils::isnan(rhs);
```
Comment (wolfv) [inline test/xsimd_test_utils.hpp]: 4 space

**Comment by wolfv** [comment]
sorry to be the styleguide guy! :)

File: test/xsimd_test_utils.hpp (hunk 4 above)
Comment (wolfv) [reply test/xsimd_test_utils.hpp]: plus bracket on newline

**Comment by serge-sans-paille** [comment]
should be okay now

**Comment by wolfv** [comment]
needs a push?

File: include/xsimd/math/xsimd_power.hpp [hunk 5]
```diff
@@ -85,6 +85,28 @@ namespace xsimd
...
+    inline typename std::enable_if<std::is_integral<T1>::value, T0>::type
+    pow(T0 const &t0, T1 const &t1)
+    {
+        T0 a = t0;
+        T1 b = t1;
+        bool const recip = b < 0;
+        T0 r{static_cast<T0>(1)};
+        while (1)
+        {
+            if (b & 1) {
```
Comment (wolfv) [inline include/xsimd/math/xsimd_power.hpp]: :octopus:

File: include/xsimd/math/xsimd_power.hpp [hunk 6]
```diff
@@ -85,6 +85,28 @@ namespace xsimd
         return detail::pow_kernel<batch<T, N>>::compute(x, y);
     }
 
+    template <class T0, class T1>
+    inline typename std::enable_if<std::is_integral<T1>::value, T0>::type
+    pow(T0 const &t0, T1 const &t1)
```
Comment (wolfv) [inline include/xsimd/math/xsimd_power.hpp]: `const T0& t0` would be preferred sry :)

**Comment by serge-sans-paille** [comment]
^^!

**Comment by wolfv** [comment]
Nice!

**Comment by serge-sans-paille** [comment]
[Brest](https://fr.wikipedia.org/wiki/Brest) !

**Pull request merged**